### PR TITLE
checkstyle_parameter

### PR DESCRIPTION
**Pull request by yichen88**

### CODE REVIEW

File: online-workflow/src/main/java/eu/itesla_project/online/tools/PrintOnlineWorkflowViolationsTool.java [hunk 1]
```diff
@@ -184,10 +184,11 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
     private void printStateStepViolations(TableFormatter formatter, Integer stateId, OnlineStep step, List<LimitViolation> violations,
                                           LimitViolationFilter violationsFilter) {
         if (violations != null) {
+            List<LimitViolation> filtedViolations = violations;
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/tools/PrintOnlineWorkflowViolationsTool.java]: FilteredViolation

File: online-workflow/src/main/java/eu/itesla_project/online/tools/PrintOnlineWorkflowPostContingencyViolationsTool.java [hunk 2]
```diff
@@ -184,10 +184,11 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
     private void printStateContingencyViolations(TableFormatter formatter, Integer stateId, String contingencyId, List<LimitViolation> violations,
                                                  LimitViolationFilter violationsFilter) {
         if (violations != null) {
+            List<LimitViolation> filtedViolations = violations;
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/tools/PrintOnlineWorkflowPostContingencyViolationsTool.java]: FilteredViolations

File: online-workflow/src/main/java/eu/itesla_project/online/security_rules/ContingencyEvaluator.java [hunk 3]
```diff
@@ -106,9 +106,6 @@ private RulesFacadeResults evaluate(String networkId, String stateId, Map<HistoD
         boolean dynamicIndexUnsafe = false;
         if (checkRules) {
             checkRules(rules, invalidRulesIndexes, bacecaseInvalidRulesIndexes);
-            if ("0".equals(stateId)) {
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/security_rules/ContingencyEvaluator.java]: Why do you remove this block?

File: online-workflow/src/main/java/eu/itesla_project/online/Utils.java [hunk 4]
```diff
@@ -136,10 +136,14 @@ public static String actionsToJsonExtended(OnlineWorkflowResults wfResults, Stri
...
+            boolean runAnalysis = false;
             if (contingencyIds == null && contingencyDb.getContingencies(network).size() == 0) {
-                contingencyIds = new HashSet<>();
+                runAnalysis = true;
             }
-            if (contingencyIds == null || !contingencyIds.isEmpty()) {
+            if (contingencyIds != null && !contingencyIds.isEmpty()) {
+                runAnalysis = true;
+            }
+            if (runAnalysis) {
```
Comment (mathbagu) [inline online-workflow/src/main/java/eu/itesla_project/online/Utils.java]: Seems obvious... can you try to write it simpler

File: online-workflow/src/main/java/eu/itesla_project/online/security_rules/ContingencyEvaluator.java (hunk 3 above)
Comment (yichen88) [reply online-workflow/src/main/java/eu/itesla_project/online/security_rules/ContingencyEvaluator.java]: Reassignement a pararmeter is not allowed and this one is not used after this statement, so i did not created a new XXX xxx = instanceFromParameter(as i do in other case)

**Pull request merged**